src/user/user-highlights.service.ts

```diff
@@ -96,11 +98,66 @@ export class UserHighlightsService {
     return new PageDto(entities, pageMetaDto);
   }
 
+  async findAllFeatured(pageOptionsDto: PageOptionsDto): Promise<PageDto<DbUserHighlight>> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.where(`user_highlights.featured = true`).orderBy("user_highlights.updated_at", "DESC");
+    queryBuilder.offset(pageOptionsDto.skip).limit(pageOptionsDto.limit);
+    const itemCount = await queryBuilder.getCount();
+    const entities = await queryBuilder.getMany();
+
+    const pageMetaDto = new PageMetaDto({ itemCount, pageOptionsDto });
+
+    return new PageDto(entities, pageMetaDto);
+  }
+
+  async addFeatured(highlightId: number, userId: number): Promise<DbUserHighlight> {
+    const user = await this.userService.findOneById(userId);
+
+    if (user.role < 100) {
+      throw new ConflictException("You are not authorized to perform this action");
+    }
+    const highlight = await this.findOneById(highlightId, userId);
+
+    if (highlight.deleted_at) {
+      throw new NotFoundException();
+    }
+
+    const updatedHighlight = await this.userHighlightRepository.save({
+      ...highlight,
+      featured: true,
+    });
+
+    return updatedHighlight;
+  }
+
+  async removeFeatured(highlightId: number, userId: number): Promise<DbUserHighlight | null> {
+    const user = await this.userService.findOneById(userId);
+
+    if (user.role < 100) {
+      throw new ConflictException("You are not authorized to perform this action");
+    }
+
+    const highlight = await this.findOneById(highlightId, userId);
+
+    if (highlight.deleted_at) {
+      throw new NotFoundException();
+    }
+
+    const updatedHighlight = await this.userHighlightRepository.save({
+      ...highlight,
+      featured: false,
+    });
```

Comment:
You can perform partial updates with `.save`

```suggestion
    const updatedHighlight = await this.userHighlightRepository.save({
      featured: false,
    });
```